SecurityException on usb

## Exception
 * __User Action:__ UI Error
 * __Request:__ Application crash
 * __Version:__ 3.5.2
 * __OS:__ Linux samsung/gts4llteusc/gts4llteusc:10/QP1A.190711.020/T837R4TYU4CTH5:user/release-keys 10 - 29
 * __Device:__ gts4llteusc
 * __Model:__ SM-T837R4
 * __Product:__ gts4llteusc Crash log

```
java.lang.RuntimeException: Unable to resume activity {com.amaze.filemanager/com.amaze.filemanager.ui.activities.MainActivity}: java.lang.SecurityException: User has not given 10281/com.amaze.filemanager permission to access device /dev/bus/usb/001/008
 at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4626)
 at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659)
 at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
 at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
 at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261)
 at android.os.Handler.dispatchMessage(Handler.java:107)
 at android.os.Looper.loop(Looper.java:237)
 at android.app.ActivityThread.main(ActivityThread.java:8107)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
 Caused by: java.lang.SecurityException: User has not given 10281/com.amaze.filemanager permission to access device /dev/bus/usb/001/008
 at android.os.Parcel.createException(Parcel.java:2088)
 at android.os.Parcel.readException(Parcel.java:2056)
 at android.os.Parcel.readException(Parcel.java:2004)
 at android.hardware.usb.IUsbSerialReader$Stub$Proxy.getSerial(IUsbSerialReader.java:123)
 at android.hardware.usb.UsbDevice.getSerialNumber(UsbDevice.java:143) at com.amaze.filemanager.utils.OTGUtil.getMassStorageDevicesConnected(OTGUtil.java:168) at com.amaze.filemanager.ui.activities.MainActivity.updateUsbInformation(MainActivity.java:1203) at com.amaze.filemanager.ui.activities.MainActivity.onResume(MainActivity.java:1195) at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1454) at android.app.Activity.performResume(Activity.java:8105) at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4616) ... 11 more Caused by: android.os.RemoteException: Remote stack trace: at com.android.server.usb.UsbUserSettingsManager.checkPermission(UsbUserSettingsManager.java:219) at com.android.server.usb.UsbSerialReader.getSerial(UsbSerialReader.java:96) at android.hardware.usb.IUsbSerialReader$Stub.onTransact(IUsbSerialReader.java:84) at android.os.Binder.execTransactInternal(Binder.java:1056) at android.os.Binder.execTransact(Binder.java:1029) ```

--- a/app/src/main/java/com/amaze/filemanager/utils/OTGUtil.java
+++ b/app/src/main/java/com/amaze/filemanager/utils/OTGUtil.java
@@ -47,7 +47,7 @@ import androidx.annotation.RequiresApi;
 import androidx.documentfile.provider.DocumentFile;
 
 /** Created by Vishal on 27-04-2017. */
-public abstract class OTGUtil {
+public class OTGUtil {
 
   private static final String TAG = OTGUtil.class.getSimpleName();
 

--- a/app/src/main/java/com/amaze/filemanager/utils/OTGUtil.java
+++ b/app/src/main/java/com/amaze/filemanager/utils/OTGUtil.java
@@ -47,7 +47,9 @@ import androidx.annotation.RequiresApi;
 import androidx.documentfile.provider.DocumentFile;
 
 /** Created by Vishal on 27-04-2017. */
-public class OTGUtil {
+public abstract class OTGUtil {
+
+  private static final String TAG = OTGUtil.class.getSimpleName();
 
   public static final String PREFIX_OTG = "otg:/";
 
@@ -163,10 +165,21 @@ public class OTGUtil {
 
       for (int i = 0; i < device.getInterfaceCount(); i++) {
         if (device.getInterface(i).getInterfaceClass() == UsbConstants.USB_CLASS_MASS_STORAGE) {
-          final @Nullable String serial =
-              Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP
-                  ? device.getSerialNumber()
-                  : null;
+          @Nullable String serial = null;
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            try {
+              serial = device.getSerialNumber();
+            } catch (SecurityException ifPermissionDenied) {
+              // May happen when device is running Android 10 or above.
+              Log.w(
+                  TAG,
+                  "Permission denied reading serial number of device "
+                      + device.getVendorId()
+                      + ":"
+                      + device.getProductId(),
+                  ifPermissionDenied);
+            }
+          }
 
           UsbOtgRepresentation usb =
               new UsbOtgRepresentation(device.getProductId(), device.getVendorId(), serial);

